TheForceEngine/TFE_DarkForces/generator.cpp

```diff
@@ -211,12 +230,13 @@ namespace TFE_DarkForces
 		return JFALSE;
 	}
 
-	Logic* obj_createGenerator(SecObject* obj, LogicSetupFunc* setupFunc, KEYWORD genType)
+	Logic* obj_createGenerator(SecObject* obj, LogicSetupFunc* setupFunc, KEYWORD genType, const char* logicName)
 	{
 		Generator* generator = (Generator*)level_alloc(sizeof(Generator));
 		memset(generator, 0, sizeof(Generator));
 
 		generator->type   = genType;
+		generator->logicName = string(logicName);
```

Comment:
I think you should change this member from "string logicName" to "char logicName[16]" and do "strncpy(generator->logicName, logicName, 15);" (or whatever the max length of a logic name is).

Reply:
Add something like this on top of this patchset, it survived a few hours of playtesting.
Don't use std::string since the only place you use this member you do ".c_str()" on it anyway.

```diff --git a/TheForceEngine/TFE_DarkForces/generator.cpp b/TheForceEngine/TFE_DarkForces/generator.cpp
index e5b9878b..83b47218 100644
--- a/TheForceEngine/TFE_DarkForces/generator.cpp
+++ b/TheForceEngine/TFE_DarkForces/generator.cpp
@@ -40,8 +40,7 @@ namespace TFE_DarkForces
                Tick       wanderTime;
                Wax*       wax;
                JBool      active;
-
-               string          logicName;              // JK: added to store a custom logic name
+               char       logicName[32];               // JK: added to store a custom logic name
        };
 
        void generatorTaskFunc(MessageType msg)
@@ -118,7 +117,7 @@ namespace TFE_DarkForces
 
                                        // Search the externally defined logics for a match
                                        TFE_ExternalData::CustomActorLogic* customLogic;
-                                       customLogic = tryFindCustomActorLogic(gen->logicName.c_str());
+                                       customLogic = tryFindCustomActorLogic(gen->logicName);
                                        if (customLogic && gameSettings->df_jsonAiLogics)
                                        {
                                                obj_setCustomActorLogic(spawn, customLogic);
@@ -236,7 +235,8 @@ namespace TFE_DarkForces
                memset(generator, 0, sizeof(Generator));
 
                generator->type   = genType;
-               generator->logicName = string(logicName);
+               memset(generator->logicName, 0, 32);
+               strncpy(generator->logicName, logicName, 31);
                generator->active = 1;
                generator->delay  = 0;
 

```

Reply:
@mlauss2 thanks for that recommendation -- I've got a couple other PRs waiting review, I'll look at this as soon as those have gone through.